sources/Outputs/src/ParGenerator.cpp

```diff
@@ -25,9 +25,9 @@ ParGenerator::write(boost::shared_ptr<parameters::ParametersSetCollection>& para
       continue;
     }
     boost::shared_ptr<parameters::ParametersSet> paramSet;
-    if (!generator.isUsingDiagram()) {
-      if (!paramSetCollection->hasParametersSet(getGeneratorParameterSetId(generator.model, DYN::doubleIsZero(generator.targetP)))) {
-        paramSet = writeConstantGeneratorsSets(activePowerCompensation, generator.model, generator.targetP, startingPointMode);
+    if (helper::generatorSharesParId(generator)) {
```

Comment:
from the point of view of behaviour, this seems correct to me. What I do not like, is that we loose the info we are using infinite diagrams here

Reply:
The issue is that the information should be held at one and one place only: it was before given in DydGenerator, ParGenerator and this test on using or not a diagramp was a 3rd assumption on what was returned in ParGenerator...